### PR TITLE
Fine-Tuning Scheduler Tutorial Updates for Lightning 2.0.x

### PR DESCRIPTION
**Pull request by speediedan**
A few minor updates to the Fine-Tuning Scheduler Tutorial to simplify it slightly and better accommodate the Lightning and PyTorch 2.0.x releases. 

### Changed

1. Support for PyTorch and PyTorch Lightning 2.0.0! Note that ``lightning>=2.0.0`` is a dependency of ``finetuning-scheduler`` 2.0.0 which this tutorial requires.
2. Use the unified `lightning` package rather than the standalone package. Fine-Tuning Scheduler (FTS) by default depends upon the `lightning` package rather than the standalone `pytorch-lightning` package beginning with FTS 2.0 (though the latter can still be installed and used similar to Lightning).
3. Leverage a feature now available in ``finetuning-scheduler`` 2.0.0 to simplify the example.
4. Updated the list of distributed strategies FTS supports as of FTS 2.x.
5. Further simplify the example by requiring torch>=1.12.1 (to avoid https://github.com/pytorch/pytorch/issues/80809 with torch 1.12.0 and the version checking/special patched version of AdamW that was required for 1.12.0)

Congrats on the successful Lightning 2.0 launch!! :rocket: :tada: 
## Before submitting

- [X] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [X] Did you make sure to **update the docs**?
- [X] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by speediedan** [comment]
@Borda @carmocca @ethanwharris @awaelchli 
Congrats on the successful 2.0 launch! Since Fine-Tuning Scheduler 2.0 was released yesterday and by default depends upon the unified `lightning` package rather than the standalone ``pytorch-lightning`` package I'm hoping this PR can be merged before too long. Only bothering you guys since I think it's important for users visiting the 2.0 tutorials to see the updated FTS default usage that aligns with the unified ``lightning`` package. Thanks again for all your work!!

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py [hunk 1]
```diff
@@ -171,33 +169,25 @@
...
 import evaluate
-import pytorch_lightning as pl
+import lightning.pytorch as pl
 import torch
 from datasets import logging as datasets_logging
-from lightning_fabric.accelerators.cuda import is_cuda_available
-from pytorch_lightning.callbacks import EarlyStopping, ModelCheckpoint
-from pytorch_lightning.loggers.tensorboard import TensorBoardLogger
-from pytorch_lightning.utilities import rank_zero_warn
+from lightning.fabric.accelerators.cuda import is_cuda_available
```
Comment (awaelchli) [inline lightning_examples/finetuning-scheduler/finetuning-scheduler.py:180]: see my comment below
```suggestion
```

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py [hunk 2]
```diff
@@ -348,6 +338,8 @@ def __init__(
                 "default".
         """
         super().__init__()
+        self.training_step_outputs = []
+        self.validation_step_outputs = []
```
Comment (awaelchli) [inline lightning_examples/finetuning-scheduler/finetuning-scheduler.py:342]: Do we actually consume these lists anywhere? I think we only append and then clear at the end of epoch. I would remove if they are redundant.

**Review by awaelchli — APPROVED** [review]
Awesome! Thanks a lot for updating

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py [hunk 3]
```diff
@@ -573,7 +544,7 @@ def train() -> None:
     trainer = pl.Trainer(
         enable_progress_bar=enable_progress_bar,
         max_epochs=100,
-        precision=16,
+        precision="16-mixed",
         accelerator="auto",
         devices=1 if is_cuda_available() else None,
```
[the remark this answers is not in the document]
Comment (speediedan) [reply lightning_examples/finetuning-scheduler/finetuning-scheduler.py:549]: Good catch!

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py (hunk 2 above)
Comment (speediedan) [reply lightning_examples/finetuning-scheduler/finetuning-scheduler.py:342]: Thanks for noticing. These were/are used in other more advanced versions of this example I run in other contexts (e.g. FSDP multi-phase scheduled fine-tuning) but I've now removed them from the basic example since they're not used as you observe.

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py [hunk 4]
```diff
@@ -171,33 +169,25 @@
...
-from typing import Any, Dict, List, Optional
-
-from packaging.version import Version
+from typing import Any, Dict, Optional
 
 import sentencepiece as sp  # noqa: F401 # isort: split
 import datasets
 import evaluate
-import pytorch_lightning as pl
+import lightning.pytorch as pl
```
[the remark this answers is not in the document]
Comment (speediedan) [reply lightning_examples/finetuning-scheduler/finetuning-scheduler.py:177]: Makes sense, nice and clean. I've updated accordingly.

File: lightning_examples/finetuning-scheduler/finetuning-scheduler.py (hunk 2 above)
Comment (awaelchli) [reply lightning_examples/finetuning-scheduler/finetuning-scheduler.py]: Very nice! I didn't know, if you want to keep them that's also fine with me.

**Review by awaelchli — APPROVED** [review]
Awesome

**Review by Borda — APPROVED** [review]
